src/scenic/core/regions.py

```diff
@@ -718,36 +740,48 @@ def _boundingPolygon(self):
         return projection
 
     @cached_property
+    @distributionFunction
     def boundingPolygon(self):
         """A PolygonalRegion bounding the mesh"""
         return PolygonalRegion(polygon=self._boundingPolygon)
 
-    def sampleGiven(self, value):
-        if isinstance(self, MeshVolumeRegion):
-            cls = MeshVolumeRegion
-        elif isinstance(self, MeshSurfaceRegion):
-            cls = MeshSurfaceRegion
-        else:
-            assert False
-
-        return cls(mesh=value[self._mesh], name=self.name, \
-            dimensions=value[self.dimensions], position=value[self.position], rotation=value[self.rotation], \
-            orientation=self.orientation, tolerance=self.tolerance, \
-            centerMesh=self.centerMesh, onDirection=self.onDirection, engine=self.engine)
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state["_mesh"] = self._mesh.copy()
```

Comment:
Presumably this is to clear the caches? It's not obvious, so I'd add a comment explaining why this is being done.